lightning/src/util/sweep.rs

```diff
@@ -922,6 +930,173 @@ where
 	}
 }
 
+/// A wrapper around [`OutputSweeper`] to be used with a sync kv store.
+pub struct OutputSweeperSyncKVStore<
+	B: Deref,
+	D: Deref,
+	E: Deref,
+	F: Deref,
+	K: Deref,
+	L: Deref,
+	O: Deref,
+> where
+	B::Target: BroadcasterInterface,
+	D::Target: ChangeDestinationSource,
+	E::Target: FeeEstimator,
+	F::Target: Filter,
+	K::Target: KVStoreSync,
+	L::Target: Logger,
+	O::Target: OutputSpender,
+{
+	sweeper: Arc<OutputSweeper<B, D, E, F, Arc<KVStoreSyncWrapper<K>>, L, O>>,
+}
+
+impl<B: Deref, D: Deref, E: Deref, F: Deref, K: Deref, L: Deref, O: Deref>
+	OutputSweeperSyncKVStore<B, D, E, F, K, L, O>
+where
+	B::Target: BroadcasterInterface,
+	D::Target: ChangeDestinationSource,
+	E::Target: FeeEstimator,
+	F::Target: Filter,
+	K::Target: KVStoreSync,
+	L::Target: Logger,
+	O::Target: OutputSpender,
+{
+	/// Constructs a new [`OutputSweeperSync`] instance.
+	pub fn new(
+		best_block: BestBlock, broadcaster: B, fee_estimator: E, chain_data_source: Option<F>,
+		output_spender: O, change_destination_source: D, kv_store: K, logger: L,
+	) -> Self {
+		let kv_store = Arc::new(KVStoreSyncWrapper(kv_store));
+
+		let sweeper = OutputSweeper::new(
+			best_block,
+			broadcaster,
+			fee_estimator,
+			chain_data_source,
+			output_spender,
+			change_destination_source,
+			kv_store,
+			logger,
+		);
+		Self { sweeper: Arc::new(sweeper) }
+	}
+
+	/// Regenerates and broadcasts the spending transaction for any outputs that are pending. Wraps
+	/// [`OutputSweeper::regenerate_and_broadcast_spend_if_necessary`].
+	pub async fn regenerate_and_broadcast_spend_if_necessary(&self) -> Result<(), ()> {
+		self.sweeper.regenerate_and_broadcast_spend_if_necessary().await
+	}
+
+	/// Wrapper around [`OutputSweeper::track_spendable_outputs`].
+	pub async fn track_spendable_outputs(
+		&self, output_descriptors: Vec<SpendableOutputDescriptor>, channel_id: Option<ChannelId>,
+		exclude_static_outputs: bool, delay_until_height: Option<u32>,
+	) -> Result<(), ()> {
+		self.sweeper
+			.track_spendable_outputs(
+				output_descriptors,
+				channel_id,
+				exclude_static_outputs,
+				delay_until_height,
+			)
+			.await
+	}
+
+	/// Returns a list of the currently tracked spendable outputs. Wraps [`OutputSweeper::tracked_spendable_outputs`].
+	pub fn tracked_spendable_outputs(&self) -> Vec<TrackedSpendableOutput> {
+		self.sweeper.tracked_spendable_outputs()
+	}
+
+	/// Wraps [`OutputSweeper::current_best_block`].
+	pub fn current_best_block(&self) -> BestBlock {
+		self.sweeper.current_best_block()
+	}
+
+	/// Returns a reference to the underlying [`OutputSweeper`].
+	pub fn sweeper_async(
```

Comment:
Needed to make this public otherwise lightning-background-processor doesn't compile.

Reply:
Then it is still publicly accessible?